inc/abstractitiltarget.class.php

```diff
@@ -1263,6 +1277,73 @@ protected function showLocationSettings($rand) {
       echo '</tr>';
    }
 
+   protected function showContractSettings($rand) {
+      global $DB;
+
+      echo '<tr>';
+      echo '<td width="15%">' . __('Contract') . '</td>';
+      echo '<td width="45%">';
+      Dropdown::showFromArray('contract_rule', static::getEnumContractRule(), [
+         'value'                 => $this->fields['contract_rule'],
+         'on_change'             => "plugin_formcreator_change_contract($rand)",
+         'rand'                  => $rand
+      ]);
+
+      echo Html::scriptBlock("plugin_formcreator_change_contract($rand)");
+      echo '</td>';
+      echo '<td width="15%">';
+      echo '<span id="contract_question_title" style="display: none">' . __('Question', 'formcreator') . '</span>';
+      echo '<span id="contract_specific_title" style="display: none">' . __('Contract ', 'formcreator') . '</span>';
+      echo '</td>';
+      echo '<td width="25%">';
+
+      echo '<div id="contract_specific_value" style="display: none">';
+      Contract::dropdown([
+         'name' => '_contract_specific',
+         'value' => $this->fields["contract_question"],
+      ]);
+      echo '</div>';
+      echo '<div id="contract_question_value" style="display: none">';
+      // select all user questions (GLPI Object)
+      $questionTable = PluginFormcreatorQuestion::getTable();
+      $sectionTable = PluginFormcreatorSection::getTable();
+      $sectionFk = PluginFormcreatorSection::getForeignKeyField();
+      $formFk = PluginFormcreatorForm::getForeignKeyField();
+      $result = $DB->request([
+         'SELECT' => [
+            $questionTable => ['id', 'name', 'values', 'itemtype'],
+            $sectionTable => ['name as sname'],
+         ],
+         'FROM' => $questionTable,
+         'INNER JOIN' => [
+            $sectionTable => [
+               'FKEY' => [
+                  $sectionTable => 'id',
+                  $questionTable => $sectionFk
+               ]
+            ],
+         ],
+         'WHERE' => [
+            "$formFk" => $this->getForm()->getID(),
+            "$questionTable.fieldtype" => 'glpiselect'
+         ]
+      ]);
+      $users_questions = [];
+      foreach ($result as $question) {
+         if ($question['itemtype'] != 'Contract') {
+            continue;
+         }
+         $users_questions[$question['sname']][$question['id']] = $question['name'];
+      }
+      Dropdown::showFromArray('_contract_question', $users_questions, [
+         'value' => $this->fields['contract_question'],
+      ]);
```

Comment:
There is a method to do such dropdown easier. Try this 

```suggestion
      PluginFormcreatorQuestion::dropdownForForm(
         $this->getForm(),
         [
            'fieldtype' => ['glpiselect'],
            'itemtype' => Contract::class
         ],
         'due_date_question',
         $this->fields['contract_question']
      );

```